src/Raven.Server/Documents/Handlers/AttachmentHandler.cs

```diff
@@ -0,0 +1,207 @@
+    {
+        [RavenAction("/databases/*/attachments", "GET")]
+        public async Task Get()
+        {
+            var documentId = GetQueryStringValueAndAssertIfSingleAndNotEmpty("id");
+            var name = GetQueryStringValueAndAssertIfSingleAndNotEmpty("name");
+
+            DocumentsOperationContext context;
+            using (ContextPool.AllocateOperationContext(out context))
+            using (context.OpenReadTransaction())
+            {
+                var attachment = Database.DocumentsStorage.GetAttachment(context, documentId, name);
+                if (attachment == null)
+                {
+                    HttpContext.Response.StatusCode = (int)HttpStatusCode.NotFound;
+                    return;
+                }
+
+                var etag = GetLongFromHeaders("If-None-Match");
+                if (etag == attachment.Etag)
+                {
+                    HttpContext.Response.StatusCode = (int)HttpStatusCode.NotModified;
+                    return;
+                }
+
+                HttpContext.Response.Headers[Constants.Headers.Etag] = "\"" + attachment.Etag + "\"";
+
+                // Ensure that files are not cached at the browser side.
+                HttpContext.Response.Headers["Cache-Control"] = "no-cache, no-store, must-revalidate";
+                HttpContext.Response.Headers["Expires"] = "0";
+
+                HttpContext.Response.Headers["Content-Type"] = attachment.ContentType.ToString();
+
+                var stream = Database.DocumentsStorage.GetAttachmentStream(context, attachment.StreamIdentifier);
+                if (stream == null)
+                {
+                    throw new FileNotFoundException("Attachment is not found. This should not happen.");
+                }
+
+                var fileName = Path.GetFileName(attachment.Name);
+                HttpContext.Response.Headers["Content-Disposition"] = $"attachment; filename=\"{fileName}\"";
+                // TODO: HttpContext.Response.Headers["Content-Range"] = ;
+
+                await stream.CopyToAsync(ResponseBodyStream());
+            }
+        }
+
+        [RavenAction("/databases/*/attachments", "PUT")]
+        public async Task Put()
+        {
+            DocumentsOperationContext context;
+            using (ContextPool.AllocateOperationContext(out context))
+            {
+                var id = GetQueryStringValueAndAssertIfSingleAndNotEmpty("id");
+                var name = GetQueryStringValueAndAssertIfSingleAndNotEmpty("name");
+                var contentType = GetStringQueryString("contentType", false) ?? "";
+
+                var tempPath = GetUniqueTempFileName(Path.GetTempPath(), "tmp");
+                using (var file = new FileStream(tempPath, FileMode.CreateNew, FileAccess.ReadWrite, FileShare.None, 4096, FileOptions.DeleteOnClose) )
+                {
+                    await RequestBodyStream().CopyToAsync(file, 81920, Database.DatabaseShutdown);
```

Comment:
This allows 80 KB, why? You can copy manually using the managed buffer from the context and save all those allocations